presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -922,7 +924,9 @@ private void commitShared()
             // Partition directory must be put back before relevant metastore operation can be undone
             committer.undoAlterPartitionOperations();
 
-            rollbackShared();
+            if (!skipStagingCleanupOnFailure) {
+                rollbackShared();
```

Comment:
Please rename it to something like `deleteStagingData`. `rollbackShared` sounds like it does something more than simply deleting the temporary data.